spec/support/server.rb

```diff
@@ -47,18 +48,35 @@ def last_received_report
       end
     end
 
+    def auth_response_code
+      @mutex.synchronize do
+        @auth_esponse_code
+      end
+    end
+
+    def auth_response_code=(code)
+      @mutex.synchronize do
+        @auth_esponse_code = code
+      end
+    end
+
     def clear!
       @mutex.synchronize do
-        @received_requests.clear
+        @auth_esponse_code = 200
+        @received_diagnose_requests.clear
       end
     end
   end
 
   set :logging, false
 
   post "/diag" do
-    DiagnoseServer.track_request(request)
+    MockServer.track_diagnose_request(request)
     status 200
     JSON.dump({ :token => "diag_support_token" })
   end
+
+  post "/1/auth" do
+    status MockServer.auth_response_code
```

Comment:
TODO: check the API key is sent to this endpoint, have an assertion about that? 🤷

Reply:
The difference between 401 and 500 is also still there. And not sure if we want to make that dynamic based on the push_api_key or be more explicit about what status code to return.